src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/caseupdate/services/CaseUpdater.java

```diff
@@ -1,35 +1,54 @@
 package uk.gov.hmcts.reform.bulkscanccdeventhandler.caseupdate.services;
 
 import org.springframework.stereotype.Service;
-import uk.gov.hmcts.reform.bulkscanccdeventhandler.common.model.in.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.caseupdate.model.in.CaseUpdate;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.caseupdate.model.out.CaseUpdateDetails;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.caseupdate.model.out.SuccessfulUpdateResponse;
 import uk.gov.hmcts.reform.bulkscanccdeventhandler.common.model.out.Address;
 import uk.gov.hmcts.reform.bulkscanccdeventhandler.common.model.out.SampleCase;
 import uk.gov.hmcts.reform.bulkscanccdeventhandler.common.utils.AddressExtractor;
 
+import static java.util.Collections.emptyList;
+
 @Service
 public class CaseUpdater {
 
+    public static final String EVENT_ID = "SAMPLE_EVENT_ID";
+
     private final AddressExtractor addressExtractor;
 
     public CaseUpdater(AddressExtractor addressExtractor) {
         this.addressExtractor = addressExtractor;
     }
 
-    public SampleCase update(SampleCase original, ExceptionRecord exceptionRecord) {
-        Address newAddress = addressExtractor.extractFrom(exceptionRecord.ocrDataFields);
+    public SuccessfulUpdateResponse update(CaseUpdate caseUpdate) {
+        Address newAddress = addressExtractor.extractFrom(caseUpdate.exceptionRecord.ocrDataFields);
+
+        SampleCase originalCase = caseUpdate.caseDetails.caseData;
 
         // This is just a sample implementation, we only overwrite the address here.
         // You'll probably update other fields and add new documents in your service case.
-        return new SampleCase(
-            original.legacyId,
-            original.firstName,
-            original.lastName,
-            original.dateOfBirth,
-            original.contactNumber,
-            original.email,
+        SampleCase newCase = new SampleCase(
+            originalCase.legacyId,
+            originalCase.firstName,
+            originalCase.lastName,
+            originalCase.dateOfBirth,
+            originalCase.contactNumber,
+            originalCase.email,
             newAddress,
-            original.scannedDocuments,
-            original.bulkScanCaseReference
+            originalCase.scannedDocuments,
+            originalCase.bulkScanCaseReference
+        );
+
+        return new SuccessfulUpdateResponse(
+            new CaseUpdateDetails(
+                // This is just a sample implementation.
+                // You can use different event IDs based on the changes made to a case...
+                EVENT_ID,
```

Comment:
should this `EVENT_ID` match with the case event in bulkscan service ccd definition?

Reply:
good point. fixed